packages/core/upload/admin/src/components/EditAssetDialog/index.js

```diff
@@ -116,6 +123,29 @@ export const EditAssetDialog = ({
     }
   };
 
+  if (folderStructureIsLoading) {
+    return (
+      <ModalLayout onClose={() => handleClose()} labelledBy="title">
+        <DialogHeader />
+        <LoadingBody minHeight="60vh" justifyContent="center" paddingTop={4} paddingBottom={4}>
+          <Loader>
+            {formatMessage({
+              id: getTrad('list.asset.load'),
+              defaultMessage: 'How do you want to upload your assets?',
```

Comment:
good catch!

Reply:
talked with @maevalienard about it, we will go with "We are loading the page. This action can take a few seconds."

Reply:
Mh, tbh I'm not convinced. Who is "we" and why "page"? I think we never use that kind of language in the admin UI. What do you simply think about "Asset is loading."? Also, I have never seen us mentioning that something takes a while and I'd skip that because it is dependent on the network and won't be true for most cases ...

Reply:
Works for me at least :)